### PR TITLE
Improved input validation for NaN

### PR DESCRIPTION
**Pull request by DalenW**
A fix for pull request #51. This should handle a broader scope of inputs, and defaults to 0. I'm not sure how to recommend changes so here's mine. Also removes unneeded code. 

Also I didn't test this, I don't have test data to run this on. But it's a simple enough change and I'm pretty sure it'll work.

### CODE REVIEW

**Comment by DalenW** [comment]
I'm dumb. I should've checked if the input was a string instead of wrapping it in a try/catch. Because if the string was `NaN`, it would've still resolved to `nan`, which doesn't actually help us here...

File: google_photos_takeout_helper/__main__.py [hunk 1]
```diff
@@ -425,6 +425,13 @@ def degToDmsRational(degFloat):
 
         return [(deg, 1), (deg_min, 1), (sec, 100)]
 
+    # converts a string input into a float. If it fails, it returns 0.0
+    def string_to_float(num):
+        if type(num) == str:
+            return 0.0
+        else:
+            return float(num)
+
```
Comment (TheLastGimbus) [inline google_photos_takeout_helper/__main__.py:434]: Can you move this inside `set_file_geo_data`, just above it's usage  and rename to `_str_to_float`? So it's more local

**Review by TheLastGimbus — APPROVED** [review]
Nice, thank you as always! I don't have courage to test it, so I'm betting that you already did 👍

File: google_photos_takeout_helper/__main__.py (hunk 1 above)
Comment (DalenW) [reply google_photos_takeout_helper/__main__.py]: Done.

**Comment by DalenW** [comment]
Made the requested changes, but I didn't test it. Like I said, I don't have test data anymore...

**Comment by TheLastGimbus** [comment]
Okay. I mean it 99.99% should work, just no hidden syntax issues - hope at least your IDE checked that 😅

**Pull request merged**